### PR TITLE
dsa-messaging

### PR DESCRIPTION
**Pull request by darmhoo**
#### What does this PR do?
- (replace me) Explain WHY this PR is required. E.g.: What feature it adds, what problem it solves, …

#### Description of Task proposed in this pull request?
- (replace me) A description of WHAT changed in the codebase. Additionally add the reasoning if it's complex or abstract. 

#### How should this be manually tested (Quality Assurance)?
- Explain how to manually test this feature. For example if changes were made in
the UI or in the API, explain where and if any specific access is needed.

#### What are the relevant pivotal tracker stories?
- [Story ID:  replace text with story id](paste Pivotal Tracker link here)

#### Any background context you want to add (Operations Impact)?
- (replace me) List the steps that must be taken for this PR to work.
E.g.: `npm install`, `Add "github_key" to environment variables, ...`

Be sure to include all systems that needs to be changed or which system is
affected by the change (Ex: Requires Styled Components to be installed and
configured in `babel.rc`, file to be updated).

#### What I have learned working on this feature: [If you don't put anything here, you are doing it wrong!]
- (replace me) I learned how to create a pull request template

### CODE REVIEW

File: app/Http/Controllers/MessageController.php [hunk 1]
```diff
@@ -42,4 +54,36 @@ public function store(Request $request)
             'ids' => $ids ?? null
         ]);
     }
+
+    public function sendDsa(Request $request){
```
Comment (teebarg) [inline app/Http/Controllers/MessageController.php:58]: rename, its not only specific to DSA

File: app/Http/Controllers/MessageController.php [hunk 2]
```diff
@@ -42,4 +54,36 @@ public function store(Request $request)
...
+        $roles = $request['roles'];
+        $staffIDs = [];
+        foreach($roles as $role){
```
Comment (teebarg) [inline app/Http/Controllers/MessageController.php:61]: why do we have roles? we are only send to a role at once

File: app/Http/Controllers/MessageController.php [hunk 3]
```diff
@@ -42,4 +54,36 @@ public function store(Request $request)
...
+            $staffs = User::where('role_id', $role)->get()->toArray();
```
Comment (teebarg) [inline app/Http/Controllers/MessageController.php:62]: when getting user, select only phone number and also add where portal access is 1

File: app/Http/Controllers/RoleController.php [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+use App\Role;
+use Illuminate\Http\Request;
+
+class RoleController extends Controller
+{
+    //
+    public function index(){
+        $roles = Role::all();
+        /** return list of states*/
```
Comment (teebarg) [inline app/Http/Controllers/RoleController.php:13]: can we remove this??

File: app/Http/Filters/MessageFilter.php [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace App\Http\Filters;
+
+use Illuminate\Database\Eloquent\Builder;
+
+class MessageFilter extends BaseFilter
+{
+    /**
+     * @param string $category
```
Comment (teebarg) [inline app/Http/Filters/MessageFilter.php:10]: category???

File: routes/api.php [hunk 6]
```diff
@@ -109,6 +110,7 @@
     Route::get('/jobs', 'ManualController@jobs');
     Route::post('/no-auth', 'ManualController@getNoAuth');
     Route::post('/contact_notification/{customer}', 'CustomerContactNotificationController@store');
+    Route::post('/send-dsa-message', 'MessageController@sendDsa');
```
Comment (teebarg) [inline routes/api.php:113]: let the endpoint be generic, we may use for other roles as well

File: app/Http/Controllers/MessageController.php (hunk 1 above)
Comment (darmhoo) [reply app/Http/Controllers/MessageController.php:58]: done

File: app/Http/Controllers/MessageController.php (hunk 3 above)
Comment (darmhoo) [reply app/Http/Controllers/MessageController.php:62]: done

File: app/Http/Filters/MessageFilter.php (hunk 5 above)
Comment (darmhoo) [reply app/Http/Filters/MessageFilter.php:10]: done

**Review by teebarg — APPROVED** [review]

**Pull request merged**